### PR TITLE
bump to a newer tendermint version

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

- brings LL to a later Tendermint

Closes: #XXX

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+      # - name: Login to DockerHub
+      #   uses: docker/login-action@v1
+      #   with:
+      #     username: ${{ secrets.DOCKERHUB_USERNAME }}
+      #     password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build Tendermint
+        run: |
+          make build-linux && cp build/tendermint DOCKER/tendermint
```
Comment (liamsi) [inline .github/workflows/docker.yml:51]: I didn't try this. Will this still work due to the renaming?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.4'
+
+      - run: echo https://github.com/tendermint/tendermint/blob/${GITHUB_REF#refs/tags/}/CHANGELOG.md#${GITHUB_REF#refs/tags/} > ../release_notes.md
```
Comment (liamsi) [inline .github/workflows/release.yml:21]: Should be renamed. In fact, we will very likely prep the first releases manually anyway though.

File: .goreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+env:
+  # Require use of Go modules.
+  - GO111MODULE=on
+
+builds:
+  - id: "Tendermint"
+    main: ./cmd/tendermint/main.go
+    ldflags:
+     - -s -w -X github.com/tendermint/tendermint/version.TMCoreSemVer={{ .Version }}
```
Comment (liamsi) [inline .goreleaser.yml:11]: Should be renamed too.

File: .goreleaser.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    env:
+      - CGO_ENABLED=0
```
Comment (liamsi) [inline .goreleaser.yml:13]: I guess our releases will use CGO to compile the go-leopard library.

File: Makefile [hunk 5]
```diff
@@ -4,7 +4,8 @@ PACKAGES=$(shell go list ./...)
 BUILDDIR ?= $(CURDIR)/build
 
 BUILD_TAGS?=tendermint
-LD_FLAGS = -X github.com/tendermint/tendermint/version.GitCommit=`git rev-parse --short=8 HEAD`
+VERSION := $(shell git describe --always)
+LD_FLAGS = -X github.com/tendermint/tendermint/version.TMCoreSemVer=$(VERSION)
```
Comment (liamsi) [inline Makefile:8]: Shouldn't this be renamed too?

File: go.mod [hunk 6]
```diff
@@ -1,22 +1,23 @@
...
 require (
 	github.com/BurntSushi/toml v0.3.1
 	github.com/ChainSafe/go-schnorrkel v0.0.0-20200405005733-88cbf1b4c40d
 	github.com/Workiva/go-datastructures v1.0.52
 	github.com/btcsuite/btcd v0.21.0-beta
 	github.com/btcsuite/btcutil v1.0.2
-	github.com/confio/ics23/go v0.0.0-20200817220745-f173e6211efb
-	github.com/cosmos/iavl v0.15.0-rc3.0.20201009144442-230e9bdf52cd
+	github.com/confio/ics23/go v0.6.3
+	github.com/cosmos/iavl v0.15.0
```
Comment (liamsi) [inline go.mod:12]: Wasn't iavl removed as a dependency from tendermint?

File: go.mod [hunk 7]
```diff
@@ -27,12 +28,12 @@ require (
...
-	github.com/spf13/cobra v1.1.0
+	github.com/spf13/cobra v1.1.1
 	github.com/spf13/viper v1.7.1
 	github.com/stretchr/testify v1.6.1
-	github.com/tendermint/tendermint v0.34.0-rc4
-	github.com/tendermint/tm-db v0.6.2
-	golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a
-	golang.org/x/net v0.0.0-20200813134508-3edf25e44fcc
-	google.golang.org/grpc v1.32.0
+	github.com/tendermint/tendermint v0.34.0
```
Comment (liamsi) [inline go.mod:34]: It's a bit odd that tendermint is a dep here. I think this is due to a test.

**Review by liamsi — COMMENTED** [review]
Thanks so much @marbar3778! Love to see that the v2 blockchain reactor received some attention recently. I'll go through all the changes shortly. Awesome work!

**Comment by liamsi** [comment]
One general remark/question: shouldn't we rather merge the changes from upstream as commits (instead of 'squashing' all changes into one commit)? It gives credit to the original authors (and git blame is more useful that way too). Also, the git histories would be more alike 🤔

File: go.mod (hunk 6 above)
Comment (marbar3778) [reply go.mod:12]: Nope it was added back for testing. We have an open issue to solve this: https://github.com/tendermint/tendermint/issues/5694

File: .github/workflows/release.yml (hunk 2 above)
Comment (marbar3778) [reply .github/workflows/release.yml:21]: will comment it out for now then.

File: go.mod (hunk 7 above)
Comment (marbar3778) [reply go.mod:34]: will groom through to find out why it's still here.

**Comment by marbar3778** [comment]
> One general remark/question: shouldn't we rather merge the changes from upstream as commits instead of 'squashing' all changes into one commit ([3707bf4](https://github.com/lazyledger/lazyledger-core/commit/3707bf4d06fe1a94130323e0ba773193158c2495))? The former would give credit to the original authors (and git blame is more useful that way too). Also, the git histories would be more alike 🤔

Oh I didnt mean for this to happen. Will undo it, it may be through a new pr though.

**Pull request closed without merging**